youtube_dl/extractor/zoom.py

```diff
@@ -1,68 +1,211 @@
+    def _get_real_webpage(self, url, base_url, video_id, url_type):
+        webpage = self._download_webpage(url, video_id, note='Downloading {0} webpage'.format(url_type))
         try:
             form = self._form_hidden_inputs('password_form', webpage)
         except ExtractorError:
-            form = None
-        if form:
-            password = self._downloader.params.get('videopassword')
-            if not password:
-                raise ExtractorError(
-                    'This video is protected by a passcode, use the --video-password option', expected=True)
-            is_meeting = form.get('useWhichPasswd') == 'meeting'
-            validation = self._download_json(
-                base_url + 'rec/validate%s_passwd' % ('_meet' if is_meeting else ''),
-                play_id, 'Validating passcode', 'Wrong passcode', data=urlencode_postdata({
-                    'id': form[('meet' if is_meeting else 'file') + 'Id'],
-                    'passwd': password,
-                    'action': form.get('action'),
-                }))
-            if not validation.get('status'):
-                raise ExtractorError(validation['errorMessage'], expected=True)
-            webpage = self._download_webpage(url, play_id)
-
-        data = self._parse_json(self._search_regex(
-            r'(?s)window\.__data__\s*=\s*({.+?});',
-            webpage, 'data'), play_id, js_to_json)
-
-        return {
-            'id': play_id,
-            'title': data['topic'],
-            'url': data['viewMp4Url'],
-            'width': int_or_none(data.get('viewResolvtionsWidth')),
-            'height': int_or_none(data.get('viewResolvtionsHeight')),
+            return webpage
+
+        password = self.get_param('videopassword')
+        if not password:
+            raise ExtractorError(
+                'This video is protected by a passcode: use the --video-password option', expected=True)
+        is_meeting = form.get('useWhichPasswd') == 'meeting'
+        validation = self._download_json(
+            base_url + 'rec/validate%s_passwd' % ('_meet' if is_meeting else ''),
+            video_id, 'Validating passcode', 'Wrong passcode', data=urlencode_postdata({
+                'id': form[('meet' if is_meeting else 'file') + 'Id'],
+                'passwd': password,
+                'action': form.get('action'),
+            }))
+        if not validation.get('status'):
+            raise ExtractorError(validation['errorMessage'], expected=True)
+        return self._download_webpage(url, video_id, note='Re-downloading {0} webpage'.format(url_type))
+
+    def _real_extract(self, url):
+        base_url, url_type, video_id = self._match_valid_url(url).group('base_url', 'type', 'id')
+        query = {}
+
+        if url_type == 'share':
+            webpage = self._get_real_webpage(url, base_url, video_id, 'share')
+            meeting_id = self._get_page_data(webpage, video_id)['meetingId']
+            redirect_path = self._download_json(
```

Comment:
if video doesn't exists like in the test https://us02web.zoom.us/rec/share/hkUk5Zxcga0nkyNGhVCRfzkA2gX_mzgS3LpTxEEWJz9Y_QpIQ4mZFOUx7KZRZDQA.9LGQBdqmDAYgiZ_8 result object will be None and accessing redirectUrl will throw a nasy exception.
maybe something like this?

```python
if download_json['result'] is not None and 'redirectUrl' in download_json['result']:
	redirect_path = download_json['result']['redirectUrl']
	url = urljoin(base_url, redirect_path)
	query['continueMode'] = 'true'
else:
	if 'errorMessage' in download_json:
		raise UnsupportedError(download_json['errorMessage'])
	else:
		raise UnsupportedError("No video found")
```